File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }

        vector<int> sCount(26, 0);
        vector<int> tCount(26, 0);

        for (int i = 0; i < s.size(); ++i) {
            sCount[s[i] - 'a']++;
            tCount[t[i] - 'a']++;
        }

        return sCount == tCount;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sChars = [ 0 for _ in range(26) ]
        tChars = [ 0 for _ in range(26) ]

        for c in s:
            sChars[ord(c) - ord('a')] += 1
        for c in t:
            tChars[ord(c) - ord('a')] += 1
        
        for i in range(len(sChars)):
            if sChars[i] != tChars[i]:
                return False

        return True